parser/util/LinkedList.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * A singly persistent linked list.
+ */
+export class LinkedList {
+  /**
+   * Constructs a new LinkedList.
+   * @param {*} head data associated with the first node in the linked list
+   * @param {LinkedList} tail a pointer to the next node in the linked list
+   */
+  constructor (head, tail) {
+    let size = tail instanceof LinkedList ? tail.size() + 1 : 0;
+    this.head = (h) => h === undefined ? head : (head = h);
```

Comment:
A couple of comments:
1. Let's use package for checking for undefines `_.isUndefined(...)` for better readability
2. Restructure this lambda function to be more easily readable:
```javascript
(h) => {
  if (! _.isUndefined(h)) {
    head = h;
  }
  return head;
}
```

Reply:
Alright.